### PR TITLE
add tests for symbol levels

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
Related to issue #11779, #10855, and PR #12710

### Summary of the issue:
PR #12710 has been suffering from scope creep, several of the problems discussed pre-existed in NVDA.

With Symbol level all:
-  '👕' symbol ("t-shirt") spoken as 't dash shirt' when moving by word or character
- Text "don't" spoken as "don tick t" when moving by word
- Symbols in NVDA speech UI (EG "spelling error" in French "Faute d'orthographe") are processed as symbols. The example "Faute d'orthographe" becomes "Faute d apostrophe orthographe", equivalent to "spelling e tick rror" in English, except that "apostrophe" is a longer word than "tick".

### Description of how this pull request fixes the issue:
Note, this PR does not introduce user visible changes. It is intended to help developers reason about the behavior of NVDA.

- Test (to demonstrate) behavior with Symbol level All and None (as the boundary cases for behavior) when moving by word, line, character.
- Test (to demonstrate) behavior with Symbol Level All when speech UI contains a symbol. For this, the System Test Spy global plugin had to be modified to alter translation strings of NVDA at runtime.

### Testing strategy:
System tests.

### Known issues with pull request:
None

### Change log entries:
None

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: tests/system/robot/notepadTests.py [hunk 1]
```diff
@@ -14,49 +17,289 @@
...
 
 
-def test_moveByWord_symbolLevelWord():
-	"""Disabled due to revert of PR #11856 is: "Speak all symbols when moving by words (#11779)
+class Move(_enum.Enum):
+	"""Gestures to move by different amounts"""
+	CHAR = "numpad3"
+	WORD = "numpad6"
+	LINE = "numpad9"
+	HOME = "control+home"
```
Comment (lukaszgo1) [inline tests/system/robot/notepadTests.py:32]: Since you're using review cursor for navigation it would be much clearer to use Shift+numpad 7 here. Currently this relies on the fact that review cursor follows caret and therefore it is moved to the top of the document and it is better to be explicit about the fact that you want to move review cursor rather than system caret.

**Comment by feerrenrut** [comment]
We may want to add a test for symbols in table headers, see https://github.com/nvaccess/nvda/pull/12710#issuecomment-1031277294

**Comment by feerrenrut** [comment]
We may also want to add tests for selecting text by word, see https://github.com/nvaccess/nvda/issues/11779#issuecomment-882307932

**Review by seanbudd — DISMISSED** [review]
LGTM, do we have issues for the other test cases you want to cover? Or should they go in this PR?

**Comment by feerrenrut** [comment]
After looking over #11779 and #10855 again, I think we should extend the move by word test to explicitly demonstrate multiple symbols combined, ie two bar symbols: `||`

I'll also look at adding tests for the selection and table cases. The table case will need to be a Chrome test.

**Comment by feerrenrut** [comment]
I have added the extra test scenarios.

File: tests/system/robot/symbolPronunciationTests.py [hunk 2]
```diff
@@ -535,3 +538,76 @@ def _doTest(
 		[endReached, ],
 		msg=f"End reached failure. actual vs expected. With symbolLevel {symbolLevel}"
 	)
+
+
+def test_tableHeaders():
+	"""
+	Announce the correct line when placed at the end of a link at the end of a list item in a contenteditable
```
Comment (seanbudd) [inline tests/system/robot/symbolPronunciationTests.py:545]: This is a fairly specific scenario. It would be good to have a "why" for why this behaviour is being tested, like the other tests.

**Review by seanbudd — COMMENTED** [review]
LGTM other than the following addition. Due to the size of the PR, I plan to give it another read over just to see if anything else needs clarification.

File: tests/system/robot/symbolPronunciationTests.py (hunk 2 above)
Comment (feerrenrut) [reply tests/system/robot/symbolPronunciationTests.py]: Ah yep, I forgot to go back to this. The table headers is an interesting example of "contextual" / secondary content. I.E. the table header text is still sourced from content, but it is not the primary text that is being traversed by the caret / review cursor. It raises the question of what symbol level should be used for secondary content.

File: tests/system/robot/symbolPronunciationTests.py [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+def test_moveByWord():
+	"""Move by word with symbol level 'all' then with symbol level 'none'
+	Symbol expectations:
+	➔ - When replaced by speech "right-pointing arrow", the dash should not be removed.
+	👕 - When replaced by speech "t-shirt", the dash should not be removed, honor the replacement text for
+			symbols/punctuation.
+
+	There should not be any "empty" words. E.G. a word made of two bar chars: ||
+	Something should be reported, even at symbol level None. Possible:
+	- The names of the characters: "bar bar", what if there are many (hundreds) symbols?
```
Comment (lukaszgo1) [inline tests/system/robot/symbolPronunciationTests.py:122]: I'd say this is a solved problem since when there are more than three characters in a row we just report their count assuming that the character in question is spoken at the current symbol level. I see no reason why this should work differently when navigating by word.

Comment (feerrenrut) [reply tests/system/robot/symbolPronunciationTests.py:122]: What if they are different characters, eg `!@#$%^`

Comment (feerrenrut) [reply tests/system/robot/symbolPronunciationTests.py]: `@^*_` is a better example for notepad.exe because it is treated as a word.

Comment (lukaszgo1) [reply tests/system/robot/symbolPronunciationTests.py]: I don't see what is the problem here. If we decide to report the words at the symbol level all then each symbol should be reported to remain consistent with reporting when user sets symbol level to all manually.

**Review by seanbudd — APPROVED** [review]
Looks good
